Rebus.LegacyExtensions/LegacyCompatibility/DeleteDuplicatedLegacyMessagesIncomingStep.cs

```diff
@@ -0,0 +1,53 @@
+﻿using HermaFx;
+using Rebus.Messages;
+using Rebus.Pipeline;
+using Rebus.Transport;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rebus.LegacyCompatibility
+{
+	public class DeleteDuplicatedLegacyMessagesIncomingStep : IIncomingStep
+	{
+		private readonly ITransport _transport;
+
+
+		private FieldInfo TopicExchangeFieldField = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
Why do we need this? this is hardcoding logic to an specific transport's internal details.